### ISSUE TITLE
RouterPagerAdapter with ViewPager not working

### ISSUE THREAD

**Issue by pawarn** [bluelinelabs/Conductor#363]
I have added controllers in view pager adapter using router pager adapter.
But when my main DashBoard Activity called that time all controllers oCreateView() & onAttach() method also gets called.

Below is the code snippet.

 protected View onCreateView(@NonNull LayoutInflater inflater, @NonNull final ViewGroup container) {
        allTiles.clear();
        View view = inflater.inflate(R.layout.layout_home_controller, container, false);
        ButterKnife.bind(this, view);
        controllerMap = new HashMap<>();
        setUpControllers();
        routerPagerAdapter = new RouterPagerAdapter(this) {
            @Override
            public void configureRouter(@NonNull Router router, int position) {
                if(!router.hasRootController())
                    router.setRoot(RouterTransaction.with(controllerMap.get(position)));
            }

            @Override
            public int getCount() {
                return controllerMap.size();
            }

        };
        
        homeViewPager.addOnPageChangeListener(this);
        homeViewPager.setAdapter(routerPagerAdapter);
        selectedIndex = 0;
        EventBus.getDefault().register(this);
        Log.e("BackStackIssue", "Home Controller inside onCreateView");
        return view;
    }

    public void setUpControllers() {
        controllerMap.put(0, new DashboardController());
        AskTheExpertController controller = new AskTheExpertController();
        controller.getArgs().putInt(AskTheExpertController.CATGID, AskTheExpertController.ROOTCATGID);
        controllerMap.put(1, controller);
        controllerMap.put(2,new HarvestDetailsController());
    }

**Comment by serjooo** [comment]
> But when my main DashBoard Activity called that time all controllers oCreateView() & onAttach() method also gets called.

Hello I'm not sure what you meant by the above? Are you saying that the OnCreate & onAttach of the controllers in the pager adapter are all being called? This is normal behavior, thats how usually it works. Can you elaborate more on what is happening? Like is the first controller attaching to the view? Is it not? Does the view pager not switch through the controllers? What is the main problem?

**Comment by pawarn** [comment]
Hello serjooo,
 I have added three controllers in view pager. In those three controllers, I have called some APIs.
But when I swipe to my first view pager then other 2 viewpager onCreateView() and onAttached() method also gets executed and that's why api's also gets execute on those controllers.  I don't want to execute those APIs.  Other 2 controllers API gets execute only when user swipe to next pager in pager adapter.
